model/gen-ai/registry.yaml

```diff
@@ -426,6 +430,57 @@ groups:
         type: int
         brief: The number of dimensions the resulting output embeddings should have.
         examples: [512, 1024]
+      - id: gen_ai.retrieval.type
+        stability: development
+        type: string
+        brief: The type of retrieval operation being performed.
+        examples: ["similarity", "hybrid", "keyword"]
+      - id: gen_ai.retrieval.top_k
```

Comment:
not a strong option, but wdyt about reusing [gen_ai.request.top_k](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/registry/attributes/gen-ai.md) instead of defining a new attribute? It's possible to override brief/note when you reference the attribute on the retrieval span.

I'd  be against that if we used something like `search` or `retrieval` instead  of `gen_ai.retrieval`.